src/secure_mul/mod.rs

```diff
@@ -0,0 +1,15 @@
+use rug::Integer;
+
+pub mod clear_text;
+
+pub trait H {
```

Comment:
I'd like to get a better overall picture of the algorithm being designed here, so I can better figure out the best abstractions to be used. This is my best interpretation of what we will need, but could need adjusting, and as of now I don't have all the context I need to decide one way or another